app/models/concerns/has_user_search.rb

```diff
@@ -84,7 +84,11 @@ def self.search(params, timezone, current_full_address)
                         end
 
           end_query = if params[:end_time]
-                        Time.zone.parse("#{day_month} #{params[:end_time]}")&.utc
+                        if params[:end_time] == "0:00"
```

Comment:
I'm worried that we might get "00:00" in the future.  Can you check for both?

Reply:
It seems that the current time picker component uses 24hrs format and the first 0 for "01:00" to "09:00" will be truncated.
Any possibility that we will use another time picker component? If no, adding checking for "00:00" seems a bit too defensive.